test/uk/gov/hmrc/cgtpropertydisposals/models/Generators.scala

```diff
@@ -202,11 +202,27 @@ trait TaxEnrolmentGen { this: GenUtils =>
 
 }
 
-trait DraftReturnGen { this: GenUtils =>
+trait DraftReturnGen extends LowerPriorityDraftReturnGen { this: GenUtils =>
```

Comment:
why do you need to do this?

Reply:
if you stick an implicit generator for `DraftReturn` with all the generators of all of the case classes which extend `DraftReturn` , the compiler will get confused when you try to summon `sample[DraftReturn]`. It will see the instance of `Gen[DraftReturn]` but it will also see the other generators which somehow it pieces together to something which also looks like `Gen[DraftReturn]`. Separating out seems to make the compiler happier so they don't conflict.